Application/pages/api/new-chat.ts

```diff
@@ -0,0 +1,69 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { MongoClient } from 'mongodb';
+import { getMongoDbUri } from '@/lib/dbConfig';
+import { generateHash } from '@/utility';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (req.method === 'POST') {
+    // In JavaScript, if you try to destructure properties that don't exist on an object, those variables will just be undefined.
+    // That means that we can just not pass in channelName, adminIDs and ownerID to indicate that we're creating a DM.
+    // Since we may no longer pass an ownerID, we need to make sure that memberIDs includes the sender ID in the front end
+    // as it won't make sense to pass an ownerID for DMs. After all, who owns a DM?
+    const { channelName, memberIDs, adminIDs, ownerID } = req.body;
+    let client: MongoClient | null = null;
+
+    try {
+      client = new MongoClient(getMongoDbUri());
+      await client.connect();
+      const db = client.db('Accord');
+      const chatsCollection = db.collection("Chats");
+      
+      memberIDs.sort(); // To prevent channel duplication
+
+      // Handle both text channels and DMs
+      let channelKey, newChat;
+      if (channelName && ownerID && adminIDs) { // Creating a text channel
+        channelKey = generateHash([channelName, memberIDs]); // memberIDs includes the sender ID
+
+        // Check if a channel with the same channelKey already exists
+        const existingTextChannel = await chatsCollection.findOne({ channelKey: channelKey });
+        if (existingTextChannel) {
+          return res.status(409).json({ error: 'A channel with the same name and members already exists.' });
+        }
+
+        newChat = {
+          channelKey: channelKey,
+          channelName,
+          dateCreated: new Date(),
+          ownerID: ownerID,
+          memberIDs: memberIDs,
+          adminIDs,
+          messageHistory: [] // Initialize with an empty message history
+        };
+      } else { // Creating a DM
+        // DMs will be unique as we don't allow a duplicate DM channel.
+        // The thing that controls that is the fact that we only create a DM channel upon adding a friend.
+        channelKey = generateHash([memberIDs]);
```

Comment:
- The generateHash function is used to generate a unique identifier (channelKey) for each channel based on its properties. Ensure that the generateHash function generates a truly unique key to avoid collisions.